### DISCUSSION TITLE
host.docker.internal support (on MacOS)

### DISCUSSION THREAD

**Discussion by derkoe** [podman-desktop/podman-desktop#3368] [Q&A] [answered]
We have some applications that use `host.docker.internal` with docker-compose.yml in the configuration.

How can I support this in Podman Desktop on MacOS?

**Comment by benoitf** [comment]
Hello @derkoe 

you can use `host.containers.internal` instead for now.

and you could follow https://github.com/containers/podman/issues/19361 about having the docker hostname being supported directly

**Reply by derkoe** [reply]
> you can use `host.containers.internal` instead for 

We have people in the team using Docker - so, we have to keep `host.docker.internal` in the config files.

> https://github.com/containers/podman/issues/19361

looking forward to this

**Reply by benoitf** [reply]
Maybe add an extra host in case of podman (setting host.docker.internal using `host.containers.internal` value) (for experimenting)

**Reply by derkoe** [reply]
Yes, that would work but only with a static IP (which may change per network).

`host-gateway` is also not working in podman https://github.com/containers/podman/issues/14390

**Reply by benoitf** [reply]
for `host-gateway` it should land in Podman 4.7.0 as it seems this enhancement has been merged https://github.com/containers/common/pull/1549

**Reply by derkoe** [reply]
BTW this is not fixed in Podman 4.7 for Mac because this commit did not make it: https://github.com/containers/podman/pull/20186/commits/58b5ae22e1840cf99df74f9f626a482081cc729a

A workaround is to manually download a [new version of gvproxy](https://github.com/containers/gvisor-tap-vsock/releases) and replace it in your installation

**Comment by derkoe** [comment, marked as answer]
To summarize the comments:

- Currently, `host.docker.internal` is not supported in Podman
- There is an open feature request in Podman for this: https://github.com/containers/podman/issues/19361
- You will be able to add custom host mappings with `host-gateway` starting with Podman 4.7.0 (see https://github.com/containers/podman/issues/14390 and https://github.com/containers/common/pull/1549). This can be used to create a mapping for `host.docker.internal`
